core/arch/arm/plat-stm32mp1/shared_resources.c

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (c) 2017-2018, STMicroelectronics
+ */
+
+#include <drivers/stm32mp1_clk.h>
+#include <stm32_util.h>
+
+unsigned long stm32_clock_get_rate(unsigned long id)
+{
+	return stm32mp1_clk_get_rate(id);
```

Comment:
Why have two names and this empty wrapper? You could directly use the name `stm32_clock_get_rate()` in the driver I suppose?

Reply:
You're right regarding `_get_rate()` and `_is_clock_enabled()` but not regarding `_clock_enable()` and `_clock_disable()`. To 2 later will (soon) integrate a specific sequence related to shared clock support.
Looks like I should simply drop the indirection for the 2 former. Thanks for the feedback. I'll push a fixup.

Reply:
Sorry, I don't get why the two others are different. I think you should add the additional layer only when it becomes necessary.

Reply:
:) 
let's say this allows to introduce the **shared_resources.c** and it's handy regarding the other P-Rs I want to submit (i.e https://github.com/etienne-lms/optee_os/pull/13).

Reply:
Maybe, but it does not change the fact that I'd rather have logical and minimal commits merged into master. Not "stuff that I will probably need very soon, promised, you will understand when you see it!" ;-)
Renaming and moving some code **when it's needed** is perfectly fine. Why create a file called `shared_ressources.c` when nothing is shared yet? It is confusing. Create it when you actually have something to share.

Reply:
Ok, I get you point view. Fixing...